shared/src/commonMain/kotlin/com/example/tasky/dataSource/AgendaDataSource.kt

```diff
@@ -26,6 +30,51 @@ class AgendaDataSource(
         } catch (e: Exception) {
             if (e is CancellationException) throw e
 
+            Throwable(e)
+            ResultWrapper.Error(DataError.Remote.UNKNOWN)
+        }
+
+    suspend fun deleteEvent(eventId: String): ResultWrapper<Unit, BaseError> =
+        try {
+            val response =
+                httpClient.delete(EventPath()) {
+                    parameter("eventId", eventId)
+                }
+
+            response.toResult<Unit>()
+        } catch (e: Exception) {
+            if (e is CancellationException) throw e
+
+            Throwable(e)
+            ResultWrapper.Error(DataError.Remote.UNKNOWN)
+        }
```

Comment:
Since you'll have a lot of these API calls that can all give you similar errors, could it make sense extracting try/catch into a function? Something like this maybe?

```
suspend inline fun <reified T> safeCall(execute: () -> HttpResponse): Result<T, DataError.Network> {
    val response = try {
        execute()
    } catch (e: UnresolvedAddressException) {
        e.printStackTrace()
        return Result.Error(DataError.Network.NO_INTERNET)
    } catch (e: HttpRequestTimeoutException) {
        e.printStackTrace()
        return Result.Error(DataError.Network.REQUEST_TIMEOUT)
    } catch (e: SerializationException) {
        e.printStackTrace()
        return Result.Error(DataError.Network.SERIALIZATION)
    } catch (e: Exception) {
        if (e is CancellationException) throw e
        e.printStackTrace()
        return Result.Error(DataError.Network.UNKNOWN)
    }

    return responseToResult(response)
}

suspend inline fun <reified T> responseToResult(response: HttpResponse): Result<T, DataError.Network> {
    return when (response.status.value) {
        in 200..299 -> Result.Success(response.body<T>())
        401 -> Result.Error(DataError.Network.UNAUTHORIZED)
        408 -> Result.Error(DataError.Network.REQUEST_TIMEOUT)
        409 -> Result.Error(DataError.Network.CONFLICT)
        413 -> Result.Error(DataError.Network.PAYLOAD_TOO_LARGE)
        429 -> Result.Error(DataError.Network.TOO_MANY_REQUESTS)
        in 500..599 -> Result.Error(DataError.Network.SERVER_ERROR)
        else -> Result.Error(DataError.Network.UNKNOWN)
    }
}
```